keyvault/client/keys/src/main/java/com/azure/keyvault/keys/KeyAsyncClient.java

```diff
@@ -0,0 +1,593 @@
+     * @return A {@link Mono} containing a {@link Response} whose {@link Response#value() value} contains the {@link Key created key}.
+     */
+    public Mono<Response<Key>> createECKey(ECKeyCreateConfig ecKeyCreateConfig) {
+        KeyRequestParameters parameters = new KeyRequestParameters()
+            .kty(ecKeyCreateConfig.keyType())
+            .curve(ecKeyCreateConfig.curve())
+            .keyOps(ecKeyCreateConfig.keyOperations())
+            .keyAttributes(new KeyRequestAttributes(ecKeyCreateConfig));
+        return service.createKey(endpoint, ecKeyCreateConfig.name(), API_VERSION, ACCEPT_LANGUAGE, parameters, CONTENT_TYPE_HEADER_VALUE);
+    }
+
+    /**
+     * Imports an externally created key and stores it in key vault. The import key operation may be used to import any key type
+     * into the Azure Key Vault. If the named key already exists, Azure Key Vault creates a new version of the key. This operation requires the {@code keys/import} permission.
+     *
+     * <p><strong>Code Samples</strong></p>
+     * <p>Imports a new key into key vault. Subscribes to the call asynchronously and prints out the newly imported key details
+     * when the callback happens.</p>
+     * <pre>
+     * keyAsyncClient.importKey("keyName", jsonWebKeyToImport).subscribe(keyResponse -&gt;
+     *   System.out.printf("Key is imported with name %s and id %s \n", keyResponse.value().name(), keyResponse.value().id()));
+     * </pre>
+     *
+     * @param name The name for the imported key.
+     * @param keyMaterial The Json web key being imported.
+     * @throws HttpRequestException if {@code name} is empty string.
+     * @return A {@link Mono} containing a {@link Response} whose {@link Response#value() value} contains the {@link Key imported key}.
+     */
+    public Mono<Response<Key>> importKey(String name, JsonWebKey keyMaterial){
+        KeyImportRequestParameters parameters = new KeyImportRequestParameters().key(keyMaterial);
+        return service.importKey(endpoint, name, API_VERSION, ACCEPT_LANGUAGE, parameters, CONTENT_TYPE_HEADER_VALUE);
+    }
+
+    /**
+     * Imports an externally created key and stores it in key vault. The import key operation may be used to import any key type
+     * into the Azure Key Vault. If the named key already exists, Azure Key Vault creates a new version of the key. This operation requires the {@code keys/import} permission.
+     *
+     * <p>The {@code keyImportConfig} is required and its fields {@link KeyImportConfig#name() name} and {@link KeyImportConfig#keyMaterial() key material} cannot
+     * be null. The {@link KeyImportConfig#expires() expires} and {@link KeyImportConfig#notBefore() notBefore} values in {@code keyImportConfig}
+     * are optional. If not specified, no values are set for the fields. The {@link KeyImportConfig#enabled() enabled} field is set to true and
+     * the {@link KeyImportConfig#hsm() hsm} field is set to false by Azure Key Vault, if they are not specified.</p>
```

Comment:
Be careful to only do client property validation. Don't validate service properties, and don't document here what the service expects today (as it might change). I suspect you might be specifying too much validation.

Reply:
updated.
Removed service client validation.